SharpCompress/Writer/Zip/ZipWriter.cs

```diff
@@ -121,12 +88,13 @@ private string NormalizeFilename(string filename)
             return filename.Trim('/');
         }
 
-        private int WriteHeader(string filename, DateTime? modificationTime)
+        private int WriteHeader(string filename, DateTime? modificationTime, ZipCompressionInfo explicitCompressionInfo = null)
         {
+            var currentCompressionInfo = explicitCompressionInfo != null ? explicitCompressionInfo : this.zipCompressionInfo;

```

Comment:
I will keep one method but I agree on initializing the ZipCompressionInfo at the bottom as it is not used anywhere else during the method chain :smiley_cat: